src/SFA.DAS.EAS.Portal/Application/Commands/Cohort/CohortApprovalRequestedCommandHandler.cs

```diff
@@ -0,0 +1,62 @@
+﻿using SFA.DAS.Commitments.Api.Client.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using System;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.Types;
+
+namespace SFA.DAS.EAS.Portal.Application.Commands.Cohort
+{
+    public class CohortApprovalRequestedCommandHandler : ICommandHandler<CohortApprovalRequestedCommand>
+    {
+        private readonly IAccountDocumentService _accountsService;
+        private readonly IProviderCommitmentsApi _providerCommitmentsApi;
+
+        public CohortApprovalRequestedCommandHandler(
+            IAccountDocumentService accountsService, 
+            IProviderCommitmentsApi providerCommitmentsApi)
+        {
+            _accountsService = accountsService;
+            _providerCommitmentsApi = providerCommitmentsApi;
+        }
+
+        public async Task Handle(CohortApprovalRequestedCommand command, CancellationToken cancellationToken = default)
+        {
+            var accountDocumentTask = _accountsService.Get(command.AccountId, cancellationToken);
+            var commitmentTask = _providerCommitmentsApi.GetProviderCommitment(command.ProviderId, command.CommitmentId);
+            
+            var accountDocument = await accountDocumentTask;
+            var account = accountDocument.Account;
+            var commitment = await commitmentTask;
+
+            var cohortReference = commitment.Reference;            
+            var cohort = account.Organisations.First().Cohorts.FirstOrDefault(c => c.Id != null && c.Id.Equals(cohortReference, StringComparison.OrdinalIgnoreCase));
```

Comment:
This handler looks for the cohort in the first organisation in the collection, and then adds the cohort to the first organisation.

The source event `CohortApprovalRequestedByProvider` doesn't contain an AccountLegalEntityId, only AccountId, ProviderId and CommitmentId. Why then are we storing cohorts under (the first) Organisation?

Reply:
The `CommitmentView` returned by the `GetProviderCommitment` contains a `LegalEntityId`. If that is the same as AccountLegalEntityId, we can use that to pick which organisation to add the cohort to.

Reply:
Cohort is now saved under the correct Organisation.